validator/src/commands/wait_for_restart_window/mod.rs

```diff
@@ -68,19 +94,15 @@ pub(crate) fn command(default_args: &DefaultArgs) -> App<'_, '_> {
 }
 
 pub fn execute(matches: &ArgMatches, ledger_path: &Path) {
-    let min_idle_time = value_t_or_exit!(matches, "min_idle_time", usize);
-    let identity = pubkey_of(matches, "identity");
-    let max_delinquent_stake = value_t_or_exit!(matches, "max_delinquent_stake", u8);
-    let skip_new_snapshot_check = matches.is_present("skip_new_snapshot_check");
-    let skip_health_check = matches.is_present("skip_health_check");
+    let wait_for_restart_window_arg = WaitForRestartWindowArgs::from_clap_arg_match(matches);
 
     wait_for_restart_window(
         ledger_path,
-        identity,
-        min_idle_time,
-        max_delinquent_stake,
-        skip_new_snapshot_check,
-        skip_health_check,
+        wait_for_restart_window_arg.identity,
+        wait_for_restart_window_arg.min_idle_time,
+        wait_for_restart_window_arg.max_delinquent_stake,
+        wait_for_restart_window_arg.skip_new_snapshot_check,
+        wait_for_restart_window_arg.skip_health_check,
```

Comment:
Why not just pass the whole `WaitForRestartWindowArgs` into `wait_for_restart_window` and de-structure the struct inside that function

Reply:
I think `fn wait_for_restart_window` should have its own argument struct instead of passing user input struct. It feels a bit too coupled to me.